src/vcpkg/commands.export-port.cpp

```diff
@@ -0,0 +1,280 @@
+#include <vcpkg/base/git.h>
+#include <vcpkg/base/hash.h>
+#include <vcpkg/base/json.h>
+#include <vcpkg/base/system.debug.h>
+#include <vcpkg/base/util.h>
+
+#include <vcpkg/archives.h>
+#include <vcpkg/commands.export-port.h>
+#include <vcpkg/configuration.h>
+#include <vcpkg/input.h>
+#include <vcpkg/metrics.h>
+#include <vcpkg/paragraphs.h>
+#include <vcpkg/registries.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace
+{
+    using namespace vcpkg;
+
+    void print_success_message(Path destination,
+                               StringView port_name,
+                               Optional<Version> version,
+                               Optional<StringView> registry = nullopt)
+    {
+        if (!version)
+        {
+            msg::println(msgExportPortSuccessUnversioned, msg::path = destination, msg::package_name = port_name);
+        }
+        else if (auto r = registry.get())
+        {
+            msg::println(msgExportPortSuccessFromRegistry,
+                         msg::path = destination,
+                         msg::package_name = port_name,
+                         msg::version = version.value_or_exit(VCPKG_LINE_INFO),
+                         msg::url = *r);
+        }
+        else
+        {
+            msg::println(msgExportPortSuccess,
+                         msg::path = destination,
+                         msg::package_name = port_name,
+                         msg::version = version.value_or_exit(VCPKG_LINE_INFO));
+        }
+    }
+
+    Optional<Version> try_get_port_version(Filesystem& fs, const Path& destination)
+    {
+        auto maybe_spgh = Paragraphs::try_load_port(fs, destination);
+        if (auto spgh = maybe_spgh.get())
+        {
+            if (auto scf = spgh->get())
+            {
+                return scf->to_version();
+            }
+        }
+        return nullopt;
+    }
+
+    void copy_port_files(Filesystem& fs, StringView port_name, const Path& source, const Path& destination)
+    {
+        std::error_code ec;
```

Comment:
Does it make sense to introduce `ExpectedL` versions of the filesystem ops used here while you're here?